plugins/modules/nd_rest.py

```diff
@@ -0,0 +1,243 @@
+    method: post
+    content:
+      {
+        "displayName": "ansible_nd_rest_fabric_policies_template",
+        "fabricPolicyTemplate": {},
+        "templateType": "fabricPolicy"
+      }
+  register: create_fabric_policies_template
+
+- name: Update Fabric Policy Template on NDO using PATCH method
+  cisco.nd.nd_rest:
+    host: nd
+    username: admin
+    password: SomeSecretPassword
+    path: "/mso/api/v1/templates/{{ create_fabric_policies_template.current.templateId }}"
+    method: patch
+    content:
+      [
+        {
+          "op": "replace",
+          "path": "/fabricPolicyTemplate/template/domains",
+          "value": [
+            {
+              "name": "ansible_nd_rest_physical_domain",
+              "description": "Ansible nd_rest Physical Domain test for PATCH",
+              "pool": ""
+            }
+          ]
+        }
+      ]
+"""
+
+RETURN = r"""
+"""
+
+import json
+import os
+
+# Optional, only used for YAML validation
+try:
+    import yaml
+
+    HAS_YAML = True
+except Exception:
+    HAS_YAML = False
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, sanitize
+from ansible_collections.cisco.nd.plugins.module_utils.constants import ND_REST_KEYS_TO_SANITIZE
+from ansible.module_utils._text import to_text
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        path=dict(type="str", required=True, aliases=["uri"]),
+        method=dict(
+            type="str",
+            default="get",
+            choices=["delete", "get", "post", "put", "patch", "DELETE", "GET", "POST", "PUT", "PATCH"],
+            aliases=["action"],
+        ),
+        content=dict(type="raw", aliases=["payload"]),
+        file_path=dict(type="path", aliases=["config_file"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    content = module.params.get("content")
+    path = module.params.get("path")
+    file_path = module.params.get("config_file")
+
+    nd = NDModule(module)
+
+    # Report missing file
+    if file_path and not os.path.isfile(file_path):
+        module.fail_json(msg="Cannot find/access file '{0}'".format(file_path))
+
+    if content and isinstance(content, str):
+        try:
+            if HAS_YAML:
+                # Validate YAML/JSON string with PyYAML package
+                content = yaml.safe_load(content)
+            else:
+                # Validate JSON string only with json Python package
+                content = json.loads(content)
+        except Exception as e:
+            error_msg = (
+                "Failed to parse provided YAML/JSON payload: {0}".format(to_text(e))
+                if HAS_YAML
+                else "Failed to parse provided JSON payload: {0}. If a YAML file was provided, the PyYAML package is required to be installed.".format(
```

Comment:
Should we mention payload?
```suggestion
                else "Failed to parse provided JSON payload: {0}. If a YAML payload was provided, the PyYAML package is required to be installed.".format(
```

Reply:
yes, we should, thanks